File: Hackerrank/strange-advertising.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "strconv"
    "strings"
)

// Complete the viralAdvertising function below.
func liked(n int32) int32 {
    var result int32 = 0
    if (n == 1){
        result = 2
    } else {
        result = liked(n - 1) * 3 / 2
    }
    return result
}
func viralAdvertising(n int32) int32 {
    var result int32 = 0
    if (n == 1){
        result = 2
    } else {
        result = liked (n) + viralAdvertising(n - 1)
    }
    return result
}

func main() {
    reader := bufio.NewReaderSize(os.Stdin, 1024 * 1024)

    stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
    checkError(err)

    defer stdout.Close()

    writer := bufio.NewWriterSize(stdout, 1024 * 1024)

    nTemp, err := strconv.ParseInt(readLine(reader), 10, 64)
    checkError(err)
    n := int32(nTemp)

    result := viralAdvertising(n)

    fmt.Fprintf(writer, "%d\n", result)

    writer.Flush()
}

func readLine(reader *bufio.Reader) string {
    str, _, err := reader.ReadLine()
    if err == io.EOF {
        return ""
    }

    return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
    if err != nil {
        panic(err)
    }
}

File: Hackerrank/strange-advertising.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the viralAdvertising function below.
def liked(n):
    if n == 1 :
        return 2
    return (liked(n-1)*3//2)
def viralAdvertising(n):
    if n == 1 :
        return 2
    return(liked(n) + viralAdvertising(n-1))
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    result = viralAdvertising(n)

    fptr.write(str(result) + '\n')

    fptr.close()
